src/main.rs

```diff
@@ -1,9 +1,47 @@
+use socketcan::*;
+
+use std::str::FromStr;
+
 use std::io::{self, Write};
 use std::process::Command;
 
 fn main() {
-    create_bus("vcan0");
-    destroy_bus("vcan0");
+    create_bus("vcan1");
+    destroy_bus("vcan1");
+
+    let mut user_in = Vec::new();
+    let socket_name: String;
+    let cob_id: u32;
+    let data: u8;
+    let rtr: bool;
+    let err: bool;
+
+    //parsing the arguments into the vector user_in
+    for arg in std::env::args().skip(1) {
+        //skipping name of program, arg(1)
+        user_in.push(String::from_str(&arg).expect("error parsing argument"));
+    }
+
+    //if we want to specify the message length
+    if user_in.is_empty() || user_in.len() != 5 {
+        error_exit();
+    } else if user_in.len() == 5 {
+        socket_name = user_in[0].to_string();
+        cob_id = u32::from_str_radix(&user_in[1], 16).expect("error parsing cob_id argument");
+        data = u8::from_str_radix(&user_in[2], 16).expect("error parsing argument");
+        rtr = bool::from_str(&user_in[3]).expect("error parsing rtr argument");
+        err = bool::from_str(&user_in[4]).expect("error parsing err argument");
+
+        let cs = CANSocket::open(&socket_name).unwrap();
+        let test = [data];
```

Comment:
Ultimately the user input parts are superceded by Brian's work on the options implementation, but we will still preserve the "create_frame_send_msg", and put in the "msgProcessor" module.